internal/service/advancedclustertpf/model_ClusterDescriptionProcessArgs20240805.go

```diff
@@ -19,19 +19,31 @@ func AddAdvancedConfig(ctx context.Context, tfModel *TFModel, input *admin.Clust
 			// special behavior using -1 when it is unset by the user
 			changeStreamOptionsPreAndPostImagesExpireAfterSeconds = conversion.Pointer(-1)
 		}
+		// When MongoDBMajorVersion is not 4.4 or lower, the API response for fail_index_key_too_long will always be null, to ensure no consistency issues, we need to match the config
+		failIndexKeyTooLong := inputLegacy.GetFailIndexKeyTooLong()
+		if tfModel != nil {
+			stateConfig := tfModel.AdvancedConfiguration
+			stateConfigSDK := NewAtlasReqAdvancedConfigurationLegacy(ctx, &stateConfig, diags)
+			if diags.HasError() {
+				return
+			}
+			if stateConfigSDK != nil && stateConfigSDK.GetFailIndexKeyTooLong() != failIndexKeyTooLong {
+				failIndexKeyTooLong = stateConfigSDK.GetFailIndexKeyTooLong()
+			}
```

Comment:
In SDK v2 we use the `Getter` methods:
["fail_index_key_too_long":              p20240530.GetFailIndexKeyTooLong(),](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/c50d357b59131250a8e310560bb154454bd93162/internal/service/advancedcluster/model_advanced_cluster.go#L520)
So even when the API returns `null` it will still be set to `false`